modules/ingester/flush.go

```diff
@@ -38,9 +37,9 @@ var (
 )
 
 const (
-	// Backoff for retrying 'immediate' flushes. Only counts for queue
-	// position, not wallclock time.
-	flushBackoff = 1 * time.Second
+	initialBackoff      = 1 * time.Second
+	maxBackoff          = time.Minute
+	maxCompleteAttempts = 10
```

Comment:
10 seems like a lot.  what are our scenarios:

- corrupt wal
- some kind of temporary disk i/o error
- ??

my gut says more like 3 is good

Reply:
Was thinking of disk full scenario: the wal is good but the new complete block can't be written.  Retrying over a span of a few minutes seems that it could allow time for other operations to free up space (remove completed blocks, blocks being replayed, etc).   We can increase the initial backoff and reduce the number of attempts, and still have similar retry window, does that sound better?